LibTrixi.jl/src/api_jl.jl

```diff
@@ -46,6 +46,14 @@ function trixi_finalize_simulation_jl(simstate)
         end
     end
 
+    # In course of garbage collection, MPI might get finalized before t8code related objects.
+    # This can lead to crashes because t8code allocates MPI related objects, e.g. shared
+    # memory arrays. The workaround is to finalize T8codeMesh explicitly in advance.
```

Comment:
```suggestion
    # In course of garbage collection, MPI might get finalized before t8code related objects.
    # This can lead to crashes because t8code allocates MPI related objects, e.g. shared
    # memory arrays. The workaround is to finalize T8codeMesh explicitly in advance.
    # x-ref: https://github.com/DLR-AMR/t8code/issues/1295
    # x-ref: https://github.com/trixi-framework/libtrixi/pull/215#discussion_r1843676330
```